### DISCUSSION TITLE
Can I make a new instance of Kitty with its own configuration in a new window, preferably with a new mac dock icon as well?

### DISCUSSION THREAD

**Discussion by MoltenMonster** [kovidgoyal/kitty#7389] [Q&A]
I did a bunch of looking up and didn't find working answers. What I want is one instance of kitty with my terminal tabs and a background image, and another icon my dock with its own maximized window and no background image, for a text editor.

**Comment by tamstrup** [comment]
On macOS I do the following to open a separate kitty instance with Neovim as the editor:

1. Create a folder `Vim.app`. Inside it create a file called `Vim` and set it as executable (`chmod +x Vim`). The latter must have the same name as the `.app` bundle for this to work (but without the `.app` extension).
2. In the `Vim` file add a script that opens kitty in a specific instance group, for example:
    ```zsh
    #!/bin/zsh

    kitty -T 'Vim' \
      --single-instance \
      --instance-group Vim \
      --wait-for-single-instance-window-close \
      nvim
    ```
3. Now you can double-click (or run `open Vim.app` in the directory that contains `Vim.app`), and it will open up a new Kitty instance. Opening the application again won't create a new instance because of the options specified in the script.
4. Initially the app doesn't have an icon. An easy way of fixing this is by copying a suitable icon (with Cmd+C) from a 512 x 512 PNG, or you can do what I did and just take the MacVim icon. Do the latter by finding the MacVim.app bundle in Finder (if you have it), right click and choose Get Info. In the window that opens, click the icon and type Cmd+C. Then do the same for the new `Vim.app` bundle, but now use Cmd+V to paste the icon.

For a more advanced setup, you can use kitty's many configuration options to control the behavior of the application and map keys.
